src/CircuitBreaker.ts

```diff
@@ -24,12 +24,14 @@ export class CircuitBreaker extends EventEmitter {
     // a promise that unit tests can await on, as jest's fake timers
     // get confused with async flows
     _evaluatingPromiseHook: Promise<unknown> | null;
+    _promError: boolean;
```

Comment:
the only explanation of this approach is in the ticket:

```
Update:
Finally we’re adding a new promErr attribute to the circuitBreaker class whenever prometheus can’t be queried which will be a boolean
```

...why does not explain _why_ we go this way instead of the more obvious approach of adding a state: the reasoning or limits should be documented, typically in the commit (and possibly PR) message. So that:

* reviewers can understand what we are really trying to do and why it is done this way
* future readers better can understand the reasoning (which may at that point not be adequate anymore) and thus account for it if they want to make changes

Reply:
The PR description has been updated